platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/offline/OfflineManager.java

```diff
@@ -0,0 +1,172 @@
+package com.mapbox.mapboxsdk.offline;
+
+import android.content.Context;
+import android.os.Handler;
+import android.os.Looper;
+import android.support.annotation.NonNull;
+
+import java.io.File;
+
+/**
+ * The offline manager is the main entry point for offline-related functionality.
+ * It'll help you list and create offline regions.
+ */
+public class OfflineManager {
+
+    // Default database name and size (1GB)
+    private final static String OFFLINE_DATABASE_NAME = "mbgl-offline.db";
+    private final static long OFFLINE_MAX_CACHE_SIZE = 1024 * 1024 * 1024;
+
+    // Holds the pointer to JNI DefaultFileSource
+    private long mDefaultFileSourcePtr = 0;
+
+    // Makes sure callbacks come back to the main thread
+    private Handler handler;
+
+    // This object is implemented as a singleton
+    private static OfflineManager instance;
+
+    /*
+     * Callbacks
+     */
+
+    public interface ListOfflineRegionsCallback {
+        void onList(OfflineRegion[] offlineRegions);
+        void onError(String error);
+    }
+
+    public interface CreateOfflineRegionCallback {
+        void onCreate(OfflineRegion offlineRegion);
+        void onError(String error);
+    }
+
+    /*
+     * Constructors
+     */
+
+    private OfflineManager() {
+        // For JNI use only, use OfflineManager.getInstance() below obtain the instance
+    }
+
+    private OfflineManager(Context context) {
+        // Get a pointer to the DefaultFileSource instance
+        String assetRoot = context.getFilesDir().getAbsolutePath();
+        String cachePath = assetRoot  + File.separator + OFFLINE_DATABASE_NAME;
+        mDefaultFileSourcePtr = createDefaultFileSource(cachePath, assetRoot, OFFLINE_MAX_CACHE_SIZE);
+    }
+
+    public static synchronized OfflineManager getInstance(Context context) {
+        if (instance == null) {
+            instance = new OfflineManager(context);
+        }
+
+        return instance;
+    }
+
+    /*
+     * Access token getter/setter
+     */
+    public void setAccessToken(String accessToken) {
+        setAccessToken(mDefaultFileSourcePtr, accessToken);
+    }
+
+    public String getAccessToken() {
+        return getAccessToken(mDefaultFileSourcePtr);
+    }
+
+    private Handler getHandler() {
+        if (handler == null) {
+            handler = new Handler(Looper.getMainLooper());
+        }
+
+        return handler;
+    }
+
+    /**
+     * Retrieve all regions in the offline database.
+     *
+     * The query will be executed asynchronously and the results passed to the given
```

Comment:
You should tailor the documentation comments to be reflective of the interface exposed by the SDK bindings, rather than the core mbgl interface. So in this case, this note about threading details is not relevant; it's an implementation detail between the core interface and the SDK binding implementation.